packages/teleterm/src/ui/ClusterConnect/ClusterConnect.tsx

```diff
@@ -2,13 +2,24 @@ import React, { useState } from 'react';
 import { ClusterAdd } from './ClusterAdd';
 import { ClusterLogin } from './ClusterLogin';
 import Dialog from 'design/Dialog';
+import { useAppContext } from 'teleterm/ui/appContextProvider';
 
 export function ClusterConnect(props: ClusterConnectProps) {
   const [createdClusterUri, setCreatedClusterUri] = useState<
     string | undefined
   >();
+  const { clustersService } = useAppContext();
   const clusterUri = props.clusterUri || createdClusterUri;
 
+  function handleClusterAdd(clusterUri: string): void {
+    const cluster = clustersService.findCluster(clusterUri);
+    if (cluster?.connected) {
+      props.onSuccess(clusterUri);
+    } else {
+      setCreatedClusterUri(clusterUri);
+    }
```

Comment:
At first I was worried that it's going to interfere with my changes to handling expired certs, but I think we should be fine. I call `openClusterConnectDialog` with the cluster URI already provided, so `ClusterConnect` will render `ClusterLogin` instead.

There might be a situation where someone calls this after the certs have expired and we weren't updated about it, but I think it's unlikely. Besides, if we plan to periodically sync root clusters somehow as I mentioned on Slack today, that should make it even less likely. Plus once that implemented, your code here will elegantly handle a situation in which someone tried to add a cluster which we know that it has expired certs.